File: projects/php-sdk/src/Controllers/VisitsController.php
<?php

namespace PHP_SDK\Controllers;

use Fingerprint\ServerAPI\Api\FingerprintApi;
use Fingerprint\ServerAPI\ApiException;
use Fingerprint\ServerAPI\Configuration;
use GuzzleHttp\Client;
use GuzzleHttp\Psr7\Response;
use GuzzleHttp\Psr7\ServerRequest;
use PHP_SDK\Models\MusicianResponse;
use Psr\Http\Message\MessageInterface;

class VisitsController
{
    public function getVisits(ServerRequest $request, Response $response): MessageInterface
    {
        $queryParams = $request->getQueryParams();
        $apiKey = $queryParams['apiKey'] ?? '';
        $region = $queryParams['region'] ?? '';
        $visitorId = $queryParams['visitorId'] ?? '';

        $requestId = $queryParams['requestId'] ?? null;
        $linkedId = $queryParams['linkedId'] ?? null;
        $limit = $queryParams['limit'] ?? null;
        $paginationKey = $queryParams['paginationKey'] ?? null;
        $before = $queryParams['before'] ?? null;

        $config = Configuration::getDefaultConfiguration($apiKey, $region);
        $client = new FingerprintApi(
            new Client(),
            $config
        );

        try {
            list($model, $api_response) = $client->getVisits($visitorId, $requestId, $linkedId, $limit, $paginationKey, $before);
            $result = new MusicianResponse($api_response->getStatusCode(), $api_response, $model);
        } catch (ApiException $e) {
            $result = MusicianResponse::BuildForApiException($e);
        } catch (\Throwable $e) {
            $result = new MusicianResponse(500, $e->getMessage(), $e->getMessage());
        }

        $response->getBody()->write(json_encode($result));

        return $response->withHeader('Content-Type', 'application/json');
    }

    public function deleteVisitorData(ServerRequest $request, Response $response): MessageInterface
    {
        $queryParams = $request->getQueryParams();
        $apiKey = $queryParams['apiKey'] ?? '';
        $region = $queryParams['region'] ?? '';
        $visitorId = $queryParams['visitorId'] ?? '';

        $config = Configuration::getDefaultConfiguration($apiKey, $region);
        $client = new FingerprintApi(
            new Client(),
            $config
        );

        try {
            list($model, $api_response) = $client->deleteVisitorData($visitorId);
            $result = new MusicianResponse($api_response->getStatusCode(), $api_response, $model);
        } catch (ApiException $e) {
            $result = MusicianResponse::BuildForApiException($e);
        } catch (\Throwable $e) {
            $result = new MusicianResponse(500, $e->getMessage(), $e->getMessage());
        }

        $response->getBody()->write(json_encode($result));

        return $response->withHeader('Content-Type', 'application/json');
    }
}

File: projects/java-sdk/src/main/java/com/example/java_sdk/v4/VisitsController.java
package com.example.java_sdk.v4;

import com.example.java_sdk.v4.Utils;
import com.fingerprint.v4.api.FingerprintApi;
import com.fingerprint.v4.model.*;
import com.fingerprint.v4.sdk.ApiClient;
import com.fingerprint.v4.sdk.ApiException;
import com.fingerprint.v4.sdk.ApiResponse;
import com.fingerprint.v4.sdk.Configuration;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController("VisitsControllerV4")
@RequestMapping("/v4")
public class VisitsController {
    @GetMapping("/deleteVisitorData")
    public ResponseEntity<com.example.java_sdk.v4.MusicianResponse> deleteVisitorData(
            @RequestParam(required = false, value = "") String api_key,
            @RequestParam(required = false, value = "") String region,
            @RequestParam(required = false, value = "") String visitor_id
    ) {
        ApiClient client = Configuration.getDefaultApiClient(api_key, Utils.getRegion(region));
        FingerprintApi api = new FingerprintApi(client);
        try {
            final ApiResponse<Void> apiResponse = api.deleteVisitorDataWithHttpInfo(visitor_id);
            final int code = apiResponse.getStatusCode();
            final com.example.java_sdk.v4.MusicianResponse response = new com.example.java_sdk.v4.MusicianResponse(code, apiResponse.getData(), apiResponse.getData());
            return ResponseEntity.ok(response);
        } catch (ApiException e) {
            final com.example.java_sdk.v4.MusicianResponse response = new com.example.java_sdk.v4.MusicianResponse(e);
            return ResponseEntity.ok(response);
        }
    }
}
